dashboard/app/controllers/ai_diff_controller.rb

```diff
@@ -209,14 +210,18 @@ def curriculum_courses
     course_names = @unit_group.present? ? [@unit_group.name, @unit_group.family_name] : ([@unit&.name] if @unit.present?)
 
     course_display_name = CourseOffering.find_by(id: @unit_group&.course_version&.course_offering_id)&.display_name
+
+    student_code = get_student_code(context[:viewAsUserId] || current_user.id, @level, @unit.id) if context[:type] == SharedConstants::AI_DIFF_CONTEXT[:LEVEL]
+
```

Comment:
yay! I'm glad you could re-use this even if there's still thinking to do about how to handle latency and queries